### ISSUE TITLE
Loading from best checkpoint

### ISSUE THREAD

**Issue by philip30** [PyTorchLightning/pytorch-lightning#1395]
## 🚀 Feature
Currently Pytorch lightning uses the latest version of the model for testing. In research, we want to first load the best checkpoint and do the testing from there. Also it would be good to restart from the best checkpoint after learning rate plateau as an option.

### Motivation

We want the best model for training/testing. Also for NLP it is more natural to go to the best checkpoint and restart with decayed learning rate.

### Alternatives

Manual loading and checking of the validation value (which is against lightning principal).

**Comment by williamFalcon** [comment]
this makes sense. how do you suggest to do it?  ideally you do this in lightning

model = Model.load_from_checkpoint(PATH)
trainer.test(model)

why doesn’t this fit your use case?

**Comment by philip30** [comment]
There are 2 issues here:
- Trainer has a training loop which involves testing (test_end, test_end) which does not behave as above [it uses the latest version of the model during training]. Meaning that I have to make a separate script for testing that load this checkpoint manually.
- After loss increased in the development set, we would want to restart from best checkpoint, reduce the learning rate and restart the epoch totally from that checkpoint. Currently this can't be achieved without an external bash script that tracks the model evaluation performace and (1) kill the training if loss increased, (2) restart with decayed learning rate. Which is too much work.

At the beginning I think it is as easy as loading the checkpoint internally. But I realized that in multi GPUs setting, these parameters need to be copied to each pytorch module inside every GPU. 

In summary:
- Let's implement module.restart_from_checkpoint_(.) for pytorch lightning module.

**Comment by vedal** [comment]
Not in the plan?

**Comment by dvirginz** [comment]
This will need to have the specific checkpoint path right?
Is there a way to give it the dir path, and it will load the best ckpt based on CheckpointModel obejct logic?

> this makes sense. how do you suggest to do it? ideally you do this in lightning
> 
> model = Model.load_from_checkpoint(PATH)
> trainer.test(model)
> 
> why doesn’t this fit your use case?

**Comment by Borda** [comment]
> This will need to have the specific checkpoint path right?
> Is there a way to give it the dir path, and it will load the best ckpt based on CheckpointModel obejct logic?

yes the path is in the Checkoint

**Comment by Borda** [comment]
already solved... 🐰

**Comment by GeorgeG92** [comment]
any update on this? Do we have the option to restore best model weights currently?

**Comment by blacksnail789521** [comment]
Please consider adding the `restore_best_weights` functionality similar to Keras. Currently, this feature is not available.
